compiler/passes/errorHandling.cpp

```diff
@@ -379,16 +379,21 @@ bool ErrorHandlingVisitor::enterCallExpr(CallExpr* node) {
     BlockStmt* throwBlock = new BlockStmt();
     node->replace(throwBlock);
 
-    SymExpr*   thrownExpr  = toSymExpr(node->get(1)->remove());
-    VarSymbol* thrownError = toVarSymbol(thrownExpr->symbol());
+    SymExpr*   thrownExpr    = toSymExpr(node->get(1)->remove());
+    VarSymbol* thrownError   = toVarSymbol(thrownExpr->symbol());
 
-    throwBlock->insertAtTail(new CallExpr(gSaveLineInErrorFn,
-                                          castToError(thrownError)));
+    VarSymbol* notNilError   = newTemp("not_nil_error", dtError);
+    CallExpr*  checkNilError = new CallExpr(gChplCheckNilError,
+                                            castToError(thrownError));
+
+    throwBlock->insertAtTail(new DefExpr(notNilError));
+    throwBlock->insertAtTail(new CallExpr(PRIM_MOVE, notNilError, checkNilError));
+    throwBlock->insertAtTail(new CallExpr(gSaveLineInErrorFn, notNilError));
```

Comment:
Seeing it here, I wonder if we should merge gChplCheckNilError with gSaveLineInErrorFn, so that we only need to call one of these form the compiler? Not urgent, though. If you choose to merge them, chpl_fix_thrown_error might be a reasonable method name for it.